rfd/0194-selinux-policy.md

```diff
@@ -0,0 +1,150 @@
+will use the Teleport agent configuration file to determine how to configure some of the tunable SELinux policy contexts, such as port
+numbers, file paths, and booleans using the `semanage` tool.
+
+The policy will allow the Teleport agent processes with the `selinux` subcommand only to create or update its SELinux policy. This will be enforced by
+creating a `teleport_selinux_management` domain, which will not be allowed to be a parent of any other Teleport related domain. Otherwise an attacker could
+use the SSH service to run the `selinux` subcommand and potentially modify the policy.
+
+If the SSH service itself was allowed to manage its own SELinux policy, an attacker that compromised the SSH service could modify the policy to
+be more permissive, defeating the point of the policy in the first place.
+
+If SELinux is not present on the host, the subcommand will tell the user this and exit. If SELinux is present but the Teleport agent policy is
+not installed, the subcommand will extract the embedded policy and use `checkmodule`, `semodule_package`, and `semodule` to compile and install the policy.
+
+Installation scripts and the `teleport-update` tool will be updated to install and configure the SELinux policy with the `selinux` subcommand.
+
+SELinux has 3 modes, disabled, permissive (policy violations are only logged) and enforcing (attempts to violate the policy are denied).
+The SELinux mode will not be modified by the `selinux` subcommand, users will be responsible for enabling SELinux if they so choose.
+
+The `selinux` subcommand will have a `--dry-run` flag that will print what actions it would need to take to manage the policy if run normally and
+print the policy that would be installed based on the provided Teleport agent configuration file.
+
+### Compliance
+
+To ensure the SSH agent SELinux policy is installed and SELinux is configured to enforce it, a `--selinux` flag will be added to the `teleport start` subcommand
+of Teleport agents. If the SELinux policy is not installed or SELinux is not present and in enforcing mode and `--selinux` is passed, the Teleport agent
+will exit with an error.
+
+Additionally since the SSH service is the only Teleport agent service that will be initially supported by the policy, if the `--selinux` flag is passed
+and the Teleport agent does not have the SSH service enabled, or at least one other Teleport service is enabled then the Teleport agent will exit with an error.
+Additionally if there is a SELinux policy for Teleport agents installed but it does not match the policy that would be generated with the version of the
+policy embedded in the Teleport agent and the Teleport agent's configuration file it will notify the user and exit with an error. Users will then have to
+run the `selinux` subcommand to update the SELinux policy.
+
+### Host support
+
+Initially RHEL based distributions, primarily RHEL 7 and 8, and Rocky Linux 8 will be supported.
```

Comment:
Not a blocker but it's getting quite hard to test on Centos/RHEL 7. As the distro turned EOL, all AMIs got pulled from AWS (there's a single RHEL7 amd64 image left).